### PR TITLE
fix: respect chart cache timeout setting

### LINKED ISSUE #21635: Chart cache timeout is ignored
Chart cache timeout is ignored, I only tested with dataset cache timeout set but looking at the code I think it will be ignored irrespetive of dataset cache timeout config.

#### How to reproduce the bug

1. start redis locally, and setup redis cache in config for data
2. create any chart on 1 dataset
3. set dataset cache timeout to 120 seconds
4. set chart cache timeout to 5 seconds (from properties modal).
5. open chart, and click on update chart
6. use redis cli, to check the expiry set on the cache key using `ttl {cache_key}` -> it will be 120 seconds.
7. after a minute backend still gives cached data, and we see `cached a minute ago` tag on top right corner.

### Expected results
Chart cache timout should override dataset timeout setting, and return actual data from database after 5 seconds.

### Actual results
Chart cache timout setting is ignored, and dataset cache timeout is used for checking existing cached data..

### Environment

(please complete the following information):

- browser type and version: chrome
- superset version: latest master (last commit sha: `71bf2673071d5db6688fbaefd4457aeeae3464bb`)
- python version: `python --version`: 3.9
- node.js version: `node -v`: 16
- any feature flags active: No

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [ ] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [ ] I have reproduced the issue with at least the latest released version of superset.
- [ ] I have checked the issue tracker for the same issue and I haven't found one similar.

### PR DESCRIPTION
**Pull request by mayurnewase**
### SUMMARY

fixes: #21635

### TESTING INSTRUCTIONS

start redis locally, and setup redis cache in config for data
create any chart on 1 dataset
set dataset cache timeout to 120 seconds
set chart cache timeout to 20 seconds (from properties modal).
open chart, and click on update chart
use redis cli, to check the expiry set on the cache key using ttl {cache_key} -> ishould be around 20 seconds.
after 20 seconds, click on update chart -> should not show `cached` tag on top right corner.

### ADDITIONAL INFORMATION

- [x] Has associated issue: #21635 
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/common/query_context.py [hunk 1]
```diff
@@ -46,6 +47,7 @@ class QueryContext:
     enforce_numerical_metrics: ClassVar[bool] = True
 
     datasource: BaseDatasource
+    slice_id: Optional[int] = None
```
Comment (villebro) [inline superset/common/query_context.py:50]: Not sure if I'm missing something, but I don't see this being referenced anywhere. Shouldn't we be fine with just having `form_data` as that contains `slice_id`?

Comment (villebro) [reply superset/common/query_context.py:50]: ..I'm assuming this should be `slice: Optional[Slice]`?
```suggestion
    slice: Optional[Slice] = None
```

**Comment by 1845725202** [comment]
ok




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "apache/superset"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年9月29日(星期四) 下午2:28
***@***.***&gt;;
***@***.***&gt;;
主题:&nbsp;Re: [apache/superset] fix: respect chart cache timeout setting (PR #21637)





 
@villebro commented on this pull request.
 
 
In superset/common/query_context.py:
 &gt; @@ -46,6 +47,7 @@ class QueryContext:      enforce_numerical_metrics: ClassVar[bool] = True        datasource: BaseDatasource +    slice_id: Optional[int] = None  
..I'm assuming this should be slice: Optional[Slice]?
 ⬇️ Suggested change -    slice_id: Optional[int] = None +    slice: Optional[Slice] = None

**Review by villebro — APPROVED** [review]
LGTM, thanks for the added tests 👍

**Pull request merged**